crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -482,9 +482,23 @@ fn topological_order(
 /// Returns vector containing all pairs of indices of systems with ambiguous execution order.
 /// Systems must be topologically sorted beforehand.
 fn find_ambiguities(systems: &[impl SystemContainer]) -> Vec<(usize, usize)> {
+    let ambiguity_set_labels = systems
```

Comment:
We can cut down on duplicate hashing / allocations and reduce line noise a bit by using this impl instead:

```rust
let mut ambiguity_set_labels = HashMap::default();
for set in systems.iter().flat_map(|c| c.ambiguity_sets()) {
    let len = ambiguity_set_labels.len();
    ambiguity_set_labels.entry(set).or_insert(len);
}
```

Reply:
I guess line noise is the kind of the wrong word for whats happening. its kind of the opposite "problem" if anything (although whether or not its a problem is a matter of taste).

Reply:
Done in 0a92c55c07e8ae2e46783f0279d78e787a06fe9d.